notes/lecture22.tex

```diff
@@ -0,0 +1,205 @@
+
+
+\section{Lecture 22: Non-convex constraints} 
+
+\begin{center}
+Seyed Mehran Mirramezani and Serena Yuan
+\end{center}
+
+
+Last time we discussed optimization of a convex function $f$ over non-convex sets. A typical example for such a problem is $\min \| Ax-y\|_2^2$ subject to $\|x\|_0 \leq s$. Another type of non-convex optimization is to minimize a non-convex function over a convex set. A typical example for this kind of non-convex optimization is known as $l_0$-minimization where $\min \| x\|_0$ subject to an affine convex constraint like $Ax=y$. In sum, the $l_0$-minimization is: 
+
+$$\min_{Ax=y} \|x\|_0$$
+
+One option for solving the above non-convex optimization problem is to relax the $l_0$-objective to the convex $l_1$-objective as:
+
+$$\min_{Ax=y} \|x\|_1$$
+
+A variation of the above problem is Basis Pursuit Denoising (BPDN),
+
+\begin{align}
+\min \|x\|_1 \text{ s.t. } \|Ax - y \|_2 \leq \epsilon.
+\end{align}
+
+
+Note that under appropriate assumptions on $A$, $x$ and $y$, we observed that $l_1$-minimization still gives the right answer (RIP, restricted nullspace).
+Sparse linear regression has received a lot of attention and there are many related optimization programs like:
+BPDN (allows for noise)
+$$\min_{\|Ax-y\|_2 \leq s} \|x\|_1$$
+
+Constrained Lasso
+$$\min_{\|x\|_1 \leq \lambda }\|Ax-y\|_2^2 $$
+
+Lagrangian Lasso
+$$\min \|Ax-y\|_2^2 +  \lambda \|x\|_1$$
+
+In our setting, we have \\
+
+\centering{nonconvex constraint} 
+$$\downarrow$$
+\centering{cvx relaxation }
+$$\downarrow$$
+\centering{PGD (Projected Gradient Descent)},
+
+or more directly, nonconvex constraint $\rightarrow$ PGD. 
+
```

Comment:
Somewhere here it would be good to mention one of the takeaways from the jupyter notebook I presented in class.

While convex relaxations can be solved efficiently (e.g., using interior point methods), scaling to large problem instances is an issue. Hence it makes sense to consider first order methods such as projected gradient descent (PGD). And when we're running PGD, it's natural to ask whether we need the convex relaxation in the first place or can just directly run PGD for the non-convex set.

Reply:
I uploaded the notebook, you can find it here:

https://ee227c.github.io/code/lecture22.html